apps/block_scout_web/lib/block_scout_web/captcha_helper.ex

```diff
@@ -3,35 +3,54 @@ defmodule BlockScoutWeb.CaptchaHelper do
   A helper for CAPTCHA
   """
 
-  @callback recaptcha_passed?(String.t() | nil) :: bool
-  @spec recaptcha_passed?(String.t() | nil) :: bool
-  def recaptcha_passed?(nil), do: false
+  @spec recaptcha_passed?(%{String.t() => String.t()} | nil) :: bool
+  def recaptcha_passed?(%{"recaptcha_v3_response" => recaptcha_response}) do
+    re_captcha_v3_secret_key = Application.get_env(:block_scout_web, :recaptcha)[:v3_secret_key]
+    do_recaptcha_passed?(re_captcha_v3_secret_key, recaptcha_response)
+  end
 
-  def recaptcha_passed?(recaptcha_response) do
+  def recaptcha_passed?(%{"recaptcha_response" => recaptcha_response}) do
     re_captcha_v2_secret_key = Application.get_env(:block_scout_web, :recaptcha)[:v2_secret_key]
-    re_captcha_v3_secret_key = Application.get_env(:block_scout_web, :recaptcha)[:v3_secret_key]
-    re_captcha_secret_key = re_captcha_v2_secret_key || re_captcha_v3_secret_key
-    body = "secret=#{re_captcha_secret_key}&response=#{recaptcha_response}"
+    do_recaptcha_passed?(re_captcha_v2_secret_key, recaptcha_response)
+  end
+
+  def recaptcha_passed?(_), do: Application.get_env(:block_scout_web, :recaptcha)[:is_disabled]
+
+  def do_recaptcha_passed?(recaptcha_secret_key, recaptcha_response) do
```

Comment:
```suggestion
  defp do_recaptcha_passed?(recaptcha_secret_key, recaptcha_response) do
```

this function is not used outside the module. Lets make it private.